### PR TITLE
✨ Restore filters from settings page for feedback dataset

### LINKED ISSUE #4390: [BUG-UI/UX] Getting out of settings erases the previous filters you had selected
**Describe the bug**
While doing the Awesome Argilla datasets annotation this week, I've found that if you had some Filter or Sorting option selected, you get in the settings (in my case, to check the annotation guidelines) and you go back by pressing the "See your dataset" option, any filter or sorting option is erased, and you can loose track of the record you were annotating.

**To Reproduce**
Steps to reproduce the behaviour:
1. Pick any filter or sorting option
2. Click the settings button
3. Click "See your dataset
4. Filters and sorting options are deselected.

**Expected behavior**
One might think that the filters and sorting options shouldn't be deselected, thus one would come back to the same record that it was before.

**Environment (please complete the following information):**
pre.argilla.io

### LINKED ISSUE #4421: [BUG-UI/UX] no-warning of loosing information when leaving page after not saving edited `settings` as owner
**Describe the bug**
I just updated the annotation guidelines and pressed the leave dataset button but did not get a warning about loosing information.

**To Reproduce**
1. Go to a dataset
2. Click on settings
3. Edit the setting but don't press the save button
4. Leave the page

**Expected behavior**
I would expect to get a warning

**Screenshots**
N.A.

**Environment (please complete the following information):**
 - OS [e.g. iOS]: MaxOS
 - Browser [e.g. chrome, safari]: Firefox
 - Argilla Version [e.g. 1.0.0]: 1.20
 - ElasticSearch Version [e.g. 7.10.2]: N.A.
 - Docker Image (optional) [e.g. argilla:v1.0.0]: 1.20

**Additional context**
N.A.

### PR DESCRIPTION
**Pull request by damianpumar**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,11 @@ These are the section headers that we use:
 
 ## [Unreleased]()
 
+### Added
+
+- Restore filters from feedback dataset settings ([#4461])(https://github.com/argilla-io/argilla/pull/4461)
+- Warning on feedback dataset settings when leaving page with unsaved changes ([#4461])(https://github.com/argilla-io/argilla/pull/4461)
```
Comment (ignacioct) [inline CHANGELOG.md:22]: Really nice touch!

**Review by ignacioct — APPROVED** [review]
Not the biggest JS expert, but looks nice and works as I imagined, thanks for implementing it!

File: CHANGELOG.md (hunk 1 above)
Comment (damianpumar) [reply CHANGELOG.md:22]: Yeas!! also I fixed the issue from @davidberenstein1957

**Review by leiyre — APPROVED** [review]
👌

**Pull request merged**